lib/rfc3164.js

```diff
@@ -5,39 +5,8 @@ const stringify = require('fast-safe-stringify')
 const through2 = require('through2')
 const utils = require('./utils')
 
-let options
-
-function buildMessage (data, headerBytes) {
-  const hlen = (options.cee) ? headerBytes - 6 : headerBytes
-  let message = {}
-  if (options.messageOnly) {
-    message = data.msg.substr(0, 1024 - headerBytes)
-    return message
-  }
-
-  if (options.includeProperties.length > 0) {
-    options.includeProperties.forEach((p) => { message[p] = data[p] })
-    message = stringify(message)
-  } else {
-    message = stringify(data)
-  }
-
-  if (message.length > (1024 - hlen)) {
-    message = stringify({
-      level: data.level,
-      time: data.time,
-      msg: 'message exceeded syslog 1024 byte limit',
-      originalSize: message.length
-    })
-  }
-
-  return message
-}
-
-module.exports = function ($options) {
-  options = $options
-
-  return through2.obj(function transport (data, enc, cb) {
+function messageBuilderFactory (options) {
+  return function (data) {
```

Comment:
Can you please explain the changes here? They are really difficult to follow in a diff.

Reply:
I have extracted the logic the build the message string to write.
This logic is used in this file and in the `lib/transport` too.

The main changes are:

- removed the global `options` object (that would not work if the user set multiple transports with different configurations)
- exporting the build message function